BlazarData/src/main/java/com/hubspot/blazar/data/service/CachingMetricsService.java

```diff
@@ -0,0 +1,56 @@
+package com.hubspot.blazar.data.service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.hubspot.blazar.base.ModuleBuild;
+import com.hubspot.blazar.base.RepositoryBuild;
+
+@Singleton
```

Comment:
It is not a good practice to maintain state in a singleton. This makes your class thread unsafe.
Also in your case you cannot go away by just removing the @Singleton annotation since you expose the instance in metrics registry (through the gauges) which will be potentially used by multiple server threads that will serve requests to the health check endpoint. So you need to synchronize the methods and make the shared properties moduleBuildCountMap, repoBuildCountMap, moduleBuildCountMapLastWrite, repoBuildCountMapLastWrite volatile...or use guava cache that takes care of thread safety.